lib/components/primitive-components/Via.ts

```diff
@@ -1,12 +1,27 @@
 import { viaProps } from "@tscircuit/props"
 import { PrimitiveComponent } from "../base-components/PrimitiveComponent"
-import type { Port } from "./Port"
+import { Port } from "./Port"
+import type { LayerRef } from "circuit-json"
+import { z } from "zod"
 
 export class Via extends PrimitiveComponent<typeof viaProps> {
   pcb_via_id: string | null = null
+  source_manually_placed_via_id: string | null = null
   matchedPort: Port | null = null
   isPcbPrimitive = true
 
+  constructor(props: z.input<typeof viaProps>) {
+    super(props)
+    const layers = this._getLayers()
+    for (const layer of layers) {
+      const port = new Port({ name: layer })
+      port.matchedComponents.push(this)
+      // Restrict available layer for this port to the via layer
+      port.getAvailablePcbLayers = () => [layer] as LayerRef[]
```

Comment:
i didn't realize you were overriding this function, not acceptable, set a constructor property instead